src/client/utils/textKeys.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from 'react'
+import enTextKeys from '../../translations/en.json'
+
+const placeholderRegex = new RegExp('({[a-zA-Z0-9_]+})', 'g')
+const placeholderKeyRegex = new RegExp('([a-zA-Z0-9_]+)', 'g')
+
+interface TextKeyResolver {
+  (replacements: Replacements): string[]
+  (): string
+}
+
+type TextKeys = Record<string, string>
+type Locale = 'en' | 'en_SE' | 'en_NO' | 'sv_SE' | 'nb_NO'
+export type Replacements = Record<string, string | number>
+export type TextKeyMap = Record<string, TextKeyResolver>
+
+export const TextKeyContext = React.createContext<TextKeys | null>(null)
+
+export const replacePlaceholders = (
+  resolvedKey: string,
+  replacements?: Replacements,
+) => {
+  if (!replacements) {
+    return resolvedKey
+  }
+  const matches = resolvedKey.split(placeholderRegex).filter((value) => value)
```

Comment:
Seems to me like the `filter` function doesn't really do anything here? 🤔 Or am I missing something?

Reply:
It performs a truthiness-check on `value`, which means that all values that are `undefined`, `null` or `false` (and maybe some others, javascript is weird) get removed 🙂

Reply:
Aha, of course! 🧠